src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/ApiTests/DataApiTest.cs

```diff
@@ -870,6 +873,71 @@ public async Task Data_Put_DataValuesUpdated_ExistingValueOverwritten()
             Assert.Equal(expectedValue, instance.DataValues[expectedKey]);
         }
 
+        /// <summary>
+        /// This test tests that you can combine data values defined and configured in the applicationmetadata.json
+        /// with data values added manully by the app developer through the IInstance interface.
+        /// Combining both methods should result in the two value sets beeing merged.
+        /// The test is not ideal as it actually tests the ServiceMock and not the real implementation,
+        /// but it states that it should be possible to combine the two.
```

Comment:
This part of the test doesn't give any value in my opinion. I would just remove it. Technically the logic is already being tested by tests in Storage. As that is where the mock code comes from.

Reply:
Agree, had a discussion with @acn-sbuad yesterday and will find another way of covering this feature using the app events. The intention of the test is two folded, one part document how to add manual data values and of course to get better coverage. The last part failed in this test.